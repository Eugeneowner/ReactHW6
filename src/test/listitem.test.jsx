import ListItem from "../componets/list/listItem";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
describe("ListItem component", () => {
  it("Render with price and name", () => {
    const name = "testName";
    const price = 100;
    render(
      <ListItem
        name={name}
        price={price}
        wishList={[]}
        basketList={[]}
        id={1}
      />
    );
    expect(screen.getByText(name)).toBeInTheDocument();
    expect(screen.getByText(/100/)).toBeInTheDocument();
  });
  it("Render with button buy", () => {
    render(<ListItem wishList={[]} basketList={[]} id={1} />);
    expect(screen.getByText("Buy now")).toBeInTheDocument();
  });
  it("Check click to button buy", () => {
    const id = 1;
    const onClick = jest.fn();

    render(
      <ListItem wishList={[]} basketList={[]} id={id} handleBuy={onClick} />
    );
    const buttonBuy = screen.getByText("Buy now");
    userEvent.click(buttonBuy);
    expect(onClick).toHaveBeenCalledTimes(1);
    expect(onClick).toHaveBeenCalledWith(id);
  });
  it("Render with button buy cancel", () => {
    const id = 1;
    render(<ListItem wishList={[]} basketList={[1]} id={id} />);
    expect(screen.getByText("Buy cancel")).toBeInTheDocument();
  });
  it("Check click buy cancel", () => {
    const id = 1;
    const onClick = jest.fn();
    render(
      <ListItem
        wishList={[]}
        basketList={[1]}
        id={id}
        handleRemoveBasket={onClick}
      />
    );
    const buttonCancel = screen.getByText("Buy cancel");
    userEvent.click(buttonCancel);
    expect(onClick).toHaveBeenCalledTimes(1);
    expect(onClick).toHaveBeenCalledWith(id);
  });
  it("Render with button wish list", () => {
    const id = 1;
    render(<ListItem
         wishList={[]} 
         basketList={[]} 
         id={id}/>);
    expect(screen.getByText("Add to wish")).toBeInTheDocument();
  });
  it ("Check click to buttton wish list", () => {
    const id = 1  
    const onClick = jest.fn();
    render(
      <ListItem
        wishList={[]}
        basketList={[]}
        id={id}
        handleWish={onClick}
      />)
      const buttonWish = screen.getByText("Add to wish");
      userEvent.click(buttonWish)
      expect(onClick).toHaveBeenCalledWith(id);
      expect(onClick).toHaveBeenCalledTimes(1);
  })
  it("Render with button remove wish", () => { 
    const id = 1
    render(
      <ListItem
      wishList={[id]}
      basketList={[]}
      id={id}
    />)
    expect(screen.getByTestId("btnRemoveWish")).toBeInTheDocument();
   })
   it ("Check click to button remove wish", () => {
    const id = 1  
    const onClick = jest.fn();
    render(
      <ListItem
        wishList={[id]}
        basketList={[]}
        id={id}
        handleRemoveWish={onClick}
      />)
      const buttonRemoveWish = screen.getByTestId("btnRemoveWish");
      userEvent.click(buttonRemoveWish)
      expect(onClick).toHaveBeenCalledWith(id);
      expect(onClick).toHaveBeenCalledTimes(1);
    })
});

