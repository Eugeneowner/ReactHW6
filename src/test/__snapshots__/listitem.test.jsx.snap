// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component Check render with button buy cancel 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main_products_item"
      >
        <img
          alt=""
          class="main_products_item_img"
        />
        <div
          class="main_products_item_container"
        >
          <p
            class="main_products_item_container_name"
          />
          <p
            class="main_products_item_container_price"
          >
             $
          </p>
          <button
            class="main_products_item_container_btn"
          >
            Add to wish
          </button>
          <button
            class="main_products_item_container_btn"
          >
            Buy now
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main_products_item"
    >
      <img
        alt=""
        class="main_products_item_img"
      />
      <div
        class="main_products_item_container"
      >
        <p
          class="main_products_item_container_name"
        />
        <p
          class="main_products_item_container_price"
        >
           $
        </p>
        <button
          class="main_products_item_container_btn"
        >
          Add to wish
        </button>
        <button
          class="main_products_item_container_btn"
        >
          Buy now
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ListItem component Check snapshot with button buy 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main_products_item"
      >
        <img
          alt=""
          class="main_products_item_img"
        />
        <div
          class="main_products_item_container"
        >
          <p
            class="main_products_item_container_name"
          />
          <p
            class="main_products_item_container_price"
          >
             $
          </p>
          <button
            class="main_products_item_container_btn"
          >
            Add to wish
          </button>
          <button
            class="main_products_item_container_btn"
          >
            Buy now
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main_products_item"
    >
      <img
        alt=""
        class="main_products_item_img"
      />
      <div
        class="main_products_item_container"
      >
        <p
          class="main_products_item_container_name"
        />
        <p
          class="main_products_item_container_price"
        >
           $
        </p>
        <button
          class="main_products_item_container_btn"
        >
          Add to wish
        </button>
        <button
          class="main_products_item_container_btn"
        >
          Buy now
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ListItem component Check snapshot with price and name 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main_products_item"
      >
        <img
          alt=""
          class="main_products_item_img"
        />
        <div
          class="main_products_item_container"
        >
          <p
            class="main_products_item_container_name"
          >
            testName
          </p>
          <p
            class="main_products_item_container_price"
          >
            100
             $
          </p>
          <button
            class="main_products_item_container_btn"
          >
            Add to wish
          </button>
          <button
            class="main_products_item_container_btn"
          >
            Buy now
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main_products_item"
    >
      <img
        alt=""
        class="main_products_item_img"
      />
      <div
        class="main_products_item_container"
      >
        <p
          class="main_products_item_container_name"
        >
          testName
        </p>
        <p
          class="main_products_item_container_price"
        >
          100
           $
        </p>
        <button
          class="main_products_item_container_btn"
        >
          Add to wish
        </button>
        <button
          class="main_products_item_container_btn"
        >
          Buy now
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ListItem component Render with button remove 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main_products_item"
      >
        <img
          alt=""
          class="main_products_item_img"
        />
        <div
          class="main_products_item_container"
        >
          <p
            class="main_products_item_container_name"
          />
          <p
            class="main_products_item_container_price"
          >
             $
          </p>
          <button
            class="main_products_item_container_btn"
            data-testid="btnRemoveWish"
          >
            <img
              alt="star"
              class="main_products_item_container_btn_img"
              src="favorite.png"
            />
          </button>
          <button
            class="main_products_item_container_btn"
          >
            Buy now
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main_products_item"
    >
      <img
        alt=""
        class="main_products_item_img"
      />
      <div
        class="main_products_item_container"
      >
        <p
          class="main_products_item_container_name"
        />
        <p
          class="main_products_item_container_price"
        >
           $
        </p>
        <button
          class="main_products_item_container_btn"
          data-testid="btnRemoveWish"
        >
          <img
            alt="star"
            class="main_products_item_container_btn_img"
            src="favorite.png"
          />
        </button>
        <button
          class="main_products_item_container_btn"
        >
          Buy now
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ListItem component Render with button remove wish 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main_products_item"
      >
        <img
          alt=""
          class="main_products_item_img"
        />
        <div
          class="main_products_item_container"
        >
          <p
            class="main_products_item_container_name"
          />
          <p
            class="main_products_item_container_price"
          >
             $
          </p>
          <button
            class="main_products_item_container_btn"
            data-testid="btnRemoveWish"
          >
            <img
              alt="star"
              class="main_products_item_container_btn_img"
              src="favorite.png"
            />
          </button>
          <button
            class="main_products_item_container_btn"
          >
            Buy now
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main_products_item"
    >
      <img
        alt=""
        class="main_products_item_img"
      />
      <div
        class="main_products_item_container"
      >
        <p
          class="main_products_item_container_name"
        />
        <p
          class="main_products_item_container_price"
        >
           $
        </p>
        <button
          class="main_products_item_container_btn"
          data-testid="btnRemoveWish"
        >
          <img
            alt="star"
            class="main_products_item_container_btn_img"
            src="favorite.png"
          />
        </button>
        <button
          class="main_products_item_container_btn"
        >
          Buy now
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ListItem component Render with button wish list 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main_products_item"
      >
        <img
          alt=""
          class="main_products_item_img"
        />
        <div
          class="main_products_item_container"
        >
          <p
            class="main_products_item_container_name"
          />
          <p
            class="main_products_item_container_price"
          >
             $
          </p>
          <button
            class="main_products_item_container_btn"
          >
            Add to wish
          </button>
          <button
            class="main_products_item_container_btn"
          >
            Buy now
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main_products_item"
    >
      <img
        alt=""
        class="main_products_item_img"
      />
      <div
        class="main_products_item_container"
      >
        <p
          class="main_products_item_container_name"
        />
        <p
          class="main_products_item_container_price"
        >
           $
        </p>
        <button
          class="main_products_item_container_btn"
        >
          Add to wish
        </button>
        <button
          class="main_products_item_container_btn"
        >
          Buy now
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
